document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/accounts/AccountsManager.java

```diff
@@ -49,4 +61,45 @@ public AbridgedAccountsApi getAbridgedAccounts(String link) throws ApiErrorRespo
 
         return apiClient.abridgedAccounts().get(link).execute();
     }
+
+    /**
+     * Get company-accounts resource if exists
+     *
+     * @param link - self link for the accounts resource
+     * @return CompanyAccounts object along with the status or not found status.
+     * @throws ApiErrorResponseException
+     * @throws URIValidationException
+     */
+    public CompanyAccounts getCompanyAccounts(String link) throws ApiErrorResponseException, URIValidationException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        return apiClient.companyAccounts().get(link).execute();
+    }
+
+    /**
+     * Get smallFull resources if exits and map to SmallFull IXBRL model
+     *
+     * @param link - self link for the abridged accounts resource
+     * @return SmallFullAccountIxbrl object
+     * @throws ApiErrorResponseException
+     * @throws URIValidationException
+     */
+    public SmallFullAccountIxbrl getSmallFullAccounts(String link, Transaction transaction)
+            throws ApiErrorResponseException, URIValidationException, ServiceException {
+
+        SmallFullApiData smallFullApiData = new SmallFullApiData();
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        smallFullApiData.setPreviousPeriod(apiClient.smallFull().previousPeriod()
+                .get(new StringBuilder(link).append("/previous-period").toString()).execute());
```

Comment:
Is it really needed to use StringBuilder in here
`.get(new StringBuilder(link).append("/previous-period").toString()).execute());`?

We could do: 
`.get(link +"/previous-period" ).toString()).execute());`

Reply:
Using string concatenation would be slightly more performant because we wouldn't instantiate a `StringBuilder` each time.

Reply:
it's debatable. Personal preference really